src/core/symbology/qgsmarkersymbollayer.cpp

```diff
@@ -2651,15 +2655,70 @@ QgsSymbolLayer *QgsRasterMarkerSymbolLayer::create( const QgsStringMap &props )
   return m;
 }
 
-void QgsRasterMarkerSymbolLayer::resolvePaths( QgsStringMap &properties, const QgsPathResolver &pathResolver, bool saving )
+void QgsRasterMarkerSymbolLayer::resolvePaths( QVariantMap &properties, const QgsPathResolver &pathResolver, bool saving )
 {
-  QgsStringMap::iterator it = properties.find( QStringLiteral( "name" ) );
+  QVariantMap::iterator it = properties.find( QStringLiteral( "name" ) );
   if ( it != properties.end() )
   {
-    if ( saving )
-      it.value() = QgsSymbolLayerUtils::svgSymbolPathToName( it.value(), pathResolver );
-    else
-      it.value() = QgsSymbolLayerUtils::svgSymbolNameToPath( it.value(), pathResolver );
+    QVariant &value = it.value();
+    switch ( value.type() )
```

Comment:
Can you explain the situation here?

Reply:
fixed.
I badly looked at the original code and thought this would require recursion.